packages/client/src/comp/states/NorthDakota.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+
+import { Select } from '../util/Select'
+import { BaseInput } from '../util/Input'
+
+import { SignatureBase, StatelessInfo, NoSignature } from './Base'
+import { NorthDakotaInfo, NorthDakotaIdentityType, northDakotaIdentityType, northDakotaElectionType, NorthDakotaElectionType } from '../../common'
+
+export const NorthDakota = () => {
+  const [electionType, setElectionType] = React.useState<NorthDakotaElectionType>('Both')
+  const [idType, setIdType] = React.useState<NorthDakotaIdentityType>('Non Driver ID')
+  const [idNumber, setIdNumber] = React.useState<string>('')
+
+  const enrichValues = (baseInfo: StatelessInfo): NoSignature<NorthDakotaInfo> | null => {
+    if (!idNumber) return null
+    return {
+      ...baseInfo,
+      state: 'North Dakota',
+      electionType,
+      idType,
+      idNumber
+    }
+  }
+
+  return <SignatureBase<NorthDakotaInfo> enrichValues={enrichValues}>
+    <Select
+      label='Identification Method:'
+      style={{ marginTop: 40 }}
+      options={northDakotaIdentityType}
+      value={idType}
+      onChange={value => setIdType(value)}
+    />
+    <BaseInput
+      id='idData'
+      label={idType}
+      required={true}
+      value={idNumber}
+      onChange={e => setIdNumber(e.currentTarget.value)}
+    />
+    <Select
+      label='Election Type for Mail in Ballot'
+      options={northDakotaElectionType}
+      value={electionType}
+      onChange={e => setElectionType(e)}
```

Comment:
I think you need `e => setElectionType(e.currentTarget.value)`

Reply:
It's actually like this for `util/Select.tsx`, by default MuiCSS doesn't support `onChange` with values when using TypeScript--it doesn't have the right types to do so. I'll rename `e => setElectionType(e)` to `value => setElectionType(value)` for consistency here.

I'm thinking in opening a Issue over there but it can take some time to get an update on this since the maintainers of the package might not be the same of MuiCSS (and the last time `@types/muicss` was updated is about 6 months ago).

Reply:
Thanks for clarifying.